### PR TITLE
Memory leak fix : cleanup of event manager

### PR DESCRIPTION
**Pull request by havietisov**
### Purpose
Adds changes necessary to clean insides of event manager. 

### Notes:
Very unlikely to break stuff, but still worthy of checking. 
Normal delegates replaced with lists to avoid using reflection, because I need to know what object is actually being targeted by any event handler added.

### CODE REVIEW

**Comment by Bod9001** [comment]
Looks good  waiting on other people to review

File: UnityProject/Assets/Scripts/Core/Lifecycle/CleanupUtil.cs [hunk 1]
```diff
@@ -0,0 +1,351 @@
...
+
+		List<T> survivorList = new List<T>();
+
+		for (int i = 0, max = listInQuestion.Count; i < max; i++)
+		{
+			MonoBehaviour target = targetExtractor(listInQuestion[i]);
+
+			if (listInQuestion[i] != null && (target == null || target.gameObject.scene.buildIndex == -1))
+			{
+				survivorList.Add(listInQuestion[i]);
```
Comment (MaxIsJoe) [inline UnityProject/Assets/Scripts/Core/Lifecycle/CleanupUtil.cs:32]: We prefer to minimize nesting whenever possible, so check if the entry you're checking is null and do `continue`

File: UnityProject/Assets/Scripts/Core/Lifecycle/CleanupUtil.cs [hunk 2]
```diff
@@ -0,0 +1,351 @@
...
+			{
+				if (verbose)
+				{
+					try
+					{
+						Logger.Log("Name of leaked object : " + target.name, Category.MemoryCleanup);
+					}
+					catch (Exception ee)
+					{
+						//do nothing
```
Comment (MaxIsJoe) [inline UnityProject/Assets/Scripts/Core/Lifecycle/CleanupUtil.cs:81]: As per our [development standards](https://unitystation.github.io/unitystation/contribution-guides/Development-Standards-Guide/#2-performance-gotchas) , don't leave catches empty.

File: UnityProject/Assets/Scripts/Core/Lifecycle/Spawn.cs [hunk 3]
```diff
@@ -48,6 +48,12 @@ public static ObjectPool _ObjectPool
 	/// </summary>
 	public static readonly float DefaultScatterRadius = 0.1f;
 
+	public static void Clean()
+	{
+		Logger.Log("Removed  " + CleanupUtil.RidDictionaryOfDeadElements(nameToSpawnablePrefab) + " elements from " + nameof(Spawn) + "." + nameof(nameToSpawnablePrefab), Category.MemoryCleanup);
```
Comment (MaxIsJoe) [inline UnityProject/Assets/Scripts/Core/Lifecycle/Spawn.cs:53]: [Follow our standard to how we write logs ](https://unitystation.github.io/unitystation/contribution-guides/Development-Standards-Guide/#0-in-code-documentation-and-clarity)

**Review by MaxIsJoe — CHANGES_REQUESTED** [review]

**Comment by havietisov** [comment]
I'l just branch out entire thing from dev since people comment on changes that are already integrated

**Pull request closed without merging**